onnxruntime/core/providers/cpu/tensor/cast_op.cc

```diff
@@ -308,8 +375,18 @@ Status Cast::Compute(OpKernelContext* context) const {
     return Status::OK();
   }
 
-  utils::MLTypeCallDispatcherFromTypeList<EnabledSrcTypes> dispatcher{from};
-  dispatcher.Invoke<SrcDispatcher>(to_, *context, shape, *X, *Y);
+  if (saturate_) {
+    utils::MLTypeCallDispatcherFromTypeList<EnabledSrcTypes> dispatcher{from};
+    dispatcher.Invoke<SrcDispatcher>(to_, *context, shape, *X, *Y);
+  } else if (to_ == ONNX_NAMESPACE::TensorProto::FLOAT8E4M3FN ||
+             to_ == ONNX_NAMESPACE::TensorProto::FLOAT8E4M3FNUZ ||
+             to_ == ONNX_NAMESPACE::TensorProto::FLOAT8E5M2 ||
+             to_ == ONNX_NAMESPACE::TensorProto::FLOAT8E5M2FNUZ) {
+    utils::MLTypeCallDispatcherFromTypeList<EnabledSrcTypes> dispatcher{from};
+    dispatcher.Invoke<SrcDispatcherNoSat>(to_, *context, shape, *X, *Y);
+  } else {
+    ORT_THROW("Option saturate=0 is only available for float 8 types.");
```

Comment:
the validation of `to_` and `saturate_` already happens in the ctor right? do we also need to do it here?

Reply:
removed